presto-main/src/main/java/com/facebook/presto/sql/planner/GroupingOperationRewriter.java

```diff
@@ -40,26 +38,21 @@
 {
     private GroupingOperationRewriter() {}
 
-    public static Expression rewriteGroupingOperation(GroupingOperation expression, QuerySpecification queryNode, Analysis analysis, Optional<Symbol> groupIdSymbol)
+    public static Expression rewriteGroupingOperation(GroupingOperation expression, List<List<Expression>> groupingSets, Map<NodeRef<Expression>, FieldId> columnReferenceFields, Optional<Symbol> groupIdSymbol)
```

Comment:
Maybe we could group `List<List<Expression>> groupingSets, Map<NodeRef<Expression>, FieldId> columnReferenceFields` as some named entity?

Reply:
Not sure that buys much. This class is only used in one place for what is, essentially, an implementation detail of the SQL->IR translator. I'm only adding this change here because I plan to get rid of this in the revamp of the planner.